engine/src/main/java/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java

```diff
@@ -653,15 +655,11 @@ public void prepareNextOutputFile() throws KettleException {
       }
 
       // file is guaranteed to be in place now
-      if ( meta.getExtension().equalsIgnoreCase( "xlsx" ) ) {
-        XSSFWorkbook xssfWorkbook = new XSSFWorkbook( KettleVFS.getInputStream( data.file ) );
-        if ( meta.isStreamingData() && !meta.isTemplateEnabled() ) {
-          data.wb = new SXSSFWorkbook( xssfWorkbook, 100 );
-        } else {
-          //Initialize it later after writing header/template because SXSSFWorkbook can't read/rewrite existing data,
-          // only append.
-          data.wb = xssfWorkbook;
-        }
+      if ( XLSX.equalsIgnoreCase( meta.getExtension() ) ) {
+        // Ignore, by now, if it's to use streaming!
```

Comment:
The only problem seems related with performance, I do not know the cost of create a instance of *Workbook.

Reply:
It's not a performance issue, it's a functionality one.
An SXSSFWorkbook is created based on a previously created XSSFWorkbook; 
When we 'transform' an XSSFWorkbook into an SXSSFWorkbook, we loose the capability of reading existing data or getting information like the number of existing rows, last row, etc.
That's why the instantiation of the SXSSFWorkbook has been shifted to a later moment.